src/jit/gentree.h

```diff
@@ -574,11 +574,21 @@ struct GenTree
         return isContained() && IsCnsIntOrI() && !isUsedFromSpillTemp();
     }
 
+#if FEATURE_X87_DOUBLES
```

Comment:
Note that we have already shipped in current state on Desktop and CoreCLR.  Now tracking floating-point constants more precisely could lead to an app compat issue for reason two in your reply.

As per my understanding Jit does float const upcast/downcast during jitting not in the generated jitted code.